CRM/Extension/Manager.php

```diff
@@ -292,6 +292,13 @@ public function disable($keys) {
     // TODO: to mitigate the risk of crashing during installation, scan
     // keys/statuses/types before doing anything
 
+    sort($keys);
+    $disableRequirements = $this->findDisableRequirements($keys);
+    sort($disableRequirements); // This munges order, but makes it comparable.
+    if ($keys !== $disableRequirements) {
+      throw new CRM_Extension_Exception_DependencyException("Cannot disable extension due dependencies. Consider disabling all these: " . implode(',', $disableRequirements));
+    }
+
     foreach ($keys as $key) {
```

Comment:
FIXME: This `foreach` should use the topological ordering returned by `findDisableRequirements()`. As written, it uses the user-supplied ordering.